High resolution content disappearing bug fix (fossasia#3519)

ashutoshc8101 opened the pull request:
### Fixes #3519
Content disappering on resolution > 2400px is fixed with this PR.

##### Changes: 
- Content at home page which used to disappear at width > 2400px doesn't disappear now.

#### What was wrong?
- Width of right container on home page was only defined upto 2400px.
<img src="https://user-images.githubusercontent.com/24855641/102686262-6c5ce480-420c-11eb-8f64-500099ff512a.png" width="300px" >
- Due to lack of specified width for screens > 2400px, the right container div gets automatically enlarged.
<img src="https://user-images.githubusercontent.com/24855641/102686276-7aab0080-420c-11eb-8f46-7351b910d191.png" 
  width="400px" />

- Which results in disappearing of content for the user.

#### What I did?
- I removed the media queries specifying different widths for different screen sizes.
- Instead I used relative width for sidebar(15%) and right container(85%) with flexbox .
- Now the width changes properly with different screen sizes.

Before:
At 2401px:
![before](https://user-images.githubusercontent.com/24855641/102686392-7b906200-420d-11eb-817c-9f3adba87fb2.png)
At 4000px:
![Screenshot from 2020-12-19 15-17-29](https://user-images.githubusercontent.com/24855641/102686394-81864300-420d-11eb-9508-dd1193fbdc2c.png)

Now:
At 2401px:
![Screenshot from 2020-12-19 15-18-03](https://user-images.githubusercontent.com/24855641/102686398-8b0fab00-420d-11eb-8843-e2340730be33.png)
At 4000px:
![Screenshot from 2020-12-19 15-18-18](https://user-images.githubusercontent.com/24855641/102686404-8f3bc880-420d-11eb-96ce-63b351e8f4fd.png)

Demo Link: https://pr-3621-fossasia-susi-web-chat.surge.sh

ashutoshc8101 commented:
@akshatnitd Please review

Dilshaad21 reviewed (approved):
Looks good to me :+1:

ashutoshc8101 commented:
@Dilshaad21 I fixed that. Now right toggle button disappears when required.

Dilshaad21 commented on src/components/cms/SkillCardScrollList/SkillCard.js:
```diff
@@ -74,12 +74,31 @@ class SkillCard extends Component {
+        scrollCards = 7;
+        break;
+      case window.innerWidth >= 2100:
+        scrollCards = 6;
```
Try to make these `window.innerWidth` conditions consistent with the previous ones. As the skillcard size is around 180px.

ashutoshc8101 replied:
Sir, the skillcard size is 280px. We cannot add 280px straight to 840 because i have used relative width for sidebar. When I am adding 280px, the right arrow is getting misaligned with the cards.
I have given width = 15% for the sidebar.
So, I am using the formula `window.innerWidth * 0.15 + 10px[Margin to Sidebar] + 280px[Box size]*n[number of boxes to display]` formula to get the required width for switch statements.
As this formula calculates it value from n, therefore i replaced switch with a for loop.


Now I am gettings desired result in the webpage

ashutoshc8101 commented:
@Dilshaad21 
As width of sidebar is relative. I have now replaced  `window.innerWidth` static conditions in switch with a formula.
Formula calculates width conditions using ` window.innerWidth * 0.15 + 19px[Margin to Sidebar] + 280px[Box size]*n[number of boxes to display] `

I have given 15% width to sidebar, therefore, I am multiplying window.innerWidth with 0.15

As this formula depends on n. So, I have used a reverse for loop in place of switch.
Finally, I squashed the commits.

akshatnitd commented on src/components/cms/SkillCardScrollList/SkillCard.js:
```diff
@@ -74,27 +74,22 @@ class SkillCard extends Component {
+    // Correct positioning of Right arrow for SkillCards upto 20 is supported with this for loop.
+    const MAX_CARDS_SUPPORT = 20;
+    for (let i = MAX_CARDS_SUPPORT; i > 1; i--) {
+      if (window.innerWidth >= window.innerWidth * 0.15 + 19 + i * 280) {
```
Can you add appropriate comments to explain the formula?

ashutoshc8101 replied:
@akshatnitd Comments explaining formula added.

Dilshaad21 reviewed (approved):
LGTM

ashutoshc8101 commented:
@akshatnitd Please Review

akshatnitd reviewed (approved):
LGTM 🚀
